src/rgw/rgw_rest_s3.cc

```diff
@@ -6551,11 +6554,12 @@ int RGWSelectObj_ObjStore_S3::send_response_data(bufferlist& bl, off_t ofs, off_
     }
     i++;
   }
-  m_aws_response_handler = std::make_unique<aws_response_handler>(s,this);
+
   if (processed_size == s->obj_size) {
+    if (status >=0) {
     m_aws_response_handler.get()->init_stats_response();
-    m_aws_response_handler.get()->send_stats_response(processed_size,processed_size,m_aws_response_handler->get_sql_result().size());
-    m_aws_response_handler.get()->init_end_response();
+    m_aws_response_handler.get()->send_stats_response(processed_size,processed_size,total_bytes_returned);
```

Comment:
/processed_size/ should be also part of /aws_response_handler class/
since /m_aws_response_handler/ is been accessed, first , in /send_response_data/
it should be initialized there.
something like 
/ if(m_aws_response_handler.get() == nullptr) ... /